Content.Client/GameObjects/EntitySystems/PhotoSystem.cs

```diff
@@ -0,0 +1,202 @@
+﻿using Content.Client.GameObjects.Components.Photography;
+using Content.Client.Interfaces.GameObjects;
+using Content.Shared.GameObjects.EntitySystemMessages;
+using Content.Shared.GameObjects.EntitySystems;
+using Robust.Client.GameObjects.EntitySystems;
+using Robust.Client.Interfaces.Input;
+using Robust.Client.Player;
+using Robust.Shared.Input;
+using Robust.Shared.Interfaces.Resources;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Utility;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.PixelFormats;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Content.Client.GameObjects.EntitySystems
+{
+    /// <summary>
+    /// Manages
+    /// - Requesting photos from the server
+    /// - Caching photos to disk
+    /// - Input management for click-to-take-photo
+    /// </summary>
+    public class PhotoSystem : SharedPhotoSystem
+    {
+
+#pragma warning disable 649
+        [Dependency] private readonly IGameTiming _gameTiming = default;
+        [Dependency] private readonly IInputManager _inputManager = default;
+        [Dependency] private readonly IPlayerManager _playerManager = default;
+        [Dependency] private readonly IResourceManager _resourceManager = default;
+#pragma warning restore 649
+
+        /// <summary>
+        /// Dictionary mapping photoIds to the CancellationTokenSource for the request to download the photo.
+        /// </summary>
+        private Dictionary<string, CancellationTokenSource> _photoRequestTokens = new Dictionary<string, CancellationTokenSource>();
+
+        private InputSystem _inputSystem;
+        private PhotoSystem _photoSystem;
+        private bool _blocked;
+        private bool _down;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            IoCManager.InjectDependencies(this);
+            _inputSystem = Get<InputSystem>();
+            _photoSystem = Get<PhotoSystem>();
```

Comment:
Yeah I appear to have temporarily gone brain dead while writing that part (I think it may have moved between classes a few times, before I solidified on having PhotoSystem(s) handle all image stuff)